homeassistant/components/broadlink/updater.py

```diff
@@ -154,7 +154,7 @@ def normalize(data, previous_data):
 class BroadlinkSP1UpdateManager(BroadlinkUpdateManager):
     """Manages updates for Broadlink SP1 devices."""
 
-    async def async_fetch_data(self):
+    async def async_fetch_data(self) -> dict | None:
```

Comment:
I don't think this is the correct way to solve this one.
We should be able to keep `None` here and to alter the type hints on the parent class `BroadlinkUpdateManager` 
I've opened https://github.com/astral-sh/ruff/issues/12198

Reply:
It looks like ruff is happy to make the changes - just need to find someone to do the PR :)